mongo/job_store.go

```diff
@@ -0,0 +1,170 @@
+package mongo
+
+import (
+	"context"
+	"errors"
+
+	dpMongodb "github.com/ONSdigital/dp-mongodb"
+	dpMongoLock "github.com/ONSdigital/dp-mongodb/dplock"
+	dpMongoHealth "github.com/ONSdigital/dp-mongodb/health"
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+)
+
+// jobs collection name
+const jobsCol = "jobs"
```

Comment:
The collection name is defined in the config. Maybe we should use that instead of a hard-coded constant?

Reply:
I had a little look at doing it like that but this is similar to what you have done in the dp-image-api, which seems to save some lines of code this way.

Reply:
I've changed the code as discussed - jobs_locks is now configurable too.